app/src/main/java/com/asfoundation/wallet/ui/iab/payments/carrier/verify/CarrierVerifyFragment.kt

```diff
@@ -0,0 +1,223 @@
+package com.asfoundation.wallet.ui.iab.payments.carrier.verify
+
+import android.animation.LayoutTransition
+import android.content.pm.ActivityInfo
+import android.content.res.Configuration
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.widget.doOnTextChanged
+import com.asf.wallet.R
+import com.jakewharton.rxbinding2.view.RxView
+import dagger.android.support.DaggerFragment
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import kotlinx.android.synthetic.main.dialog_buy_buttons_payment_methods.*
+import kotlinx.android.synthetic.main.fragment_carrier_verify_phone.*
+import java.math.BigDecimal
+import java.util.*
+import javax.inject.Inject
+
+class CarrierVerifyFragment : DaggerFragment(),
+    CarrierVerifyView {
+
+  @Inject
+  lateinit var presenter: CarrierVerifyPresenter
+
+  private val phoneNumberChangedSubject = PublishSubject.create<Any>()
```

Comment:
I don't see any reason to do this. A subject won't leak as long as any subscription that uses it gets cleaned up (which is the case of the stop() on the presenter). It also cannot retain the fragment, since it doesn't have any reference to it (which is not the case with a RecyclerView adapter, for example). But let me know if you find anything that says otherwise.